src/providers/FilterProvider/FilterProvider.js

```diff
@@ -0,0 +1,55 @@
+import React, { createContext, useMemo, useCallback, useReducer, useContext } from 'react';
+import { sortActions } from '../../constants/actions';
+import { filterReducer } from '../../pages/product/reducer/filterReducer';
+
+const initialState = {
+  fastDeliveryOnly: false,
+  categories: null,
+  inStock: false,
+  sortBy: null,
+  ratedBy: null,
+  price: null,
+};
+
+const FilterContext = createContext();
+
+export const useFilter = () => useContext(FilterContext);
+
+const FilterProvider = ({ children }) => {
+  const [filteredState, dispatch] = useReducer(filterReducer, initialState);
+
+  const getSortedProducts = useCallback((productList, sortBy) => {
```

Comment:
Can you explain to me why you're using useCallback here?

Reply:
On each render, new instance of functions, objects present in the component/function are created again. **useMemo**, **useCallback** are used for **memoization**. Using useCallback, useMemo gives the previous memoized value and doesn't create new instance.